crates/sui-core/src/gateway_state.rs

```diff
@@ -537,9 +538,32 @@ where
         .await?;
 
         let owned_objects = input_objects.filter_owned_objects();
-        self.set_transaction_lock(&owned_objects, transaction.clone())
+        if let Err(err) = self
+            .set_transaction_lock(&owned_objects, transaction.clone())
             .instrument(tracing::trace_span!("db_set_transaction_lock"))
-            .await?;
+            .await
+        {
+            // This is a temporary solution to get objects out of locked state.
```

Comment:
what's the long-term solution?

Reply:
I do not know. We will need to rewrite the equivocation prevention functionality, and with great effort put into designing it right.